src/client/NextApi.Client/NextApiClient.cs

```diff
@@ -303,30 +316,40 @@ private async Task<T> InvokeHttp<T>(NextApiCommand command)
                 }
             }
 
-            switch (httpSerializationType)
+            var data = await response.Content.ReadAsStringAsync();
+            var result =
+                JsonConvert.DeserializeObject<NextApiResponseJsonWrapper<T>>(data, SerializationUtils.GetJsonConfig());
+
+            if (!result.Success)
             {
-                case SerializationType.Json:
-                {
-                    // process as normal nextapi response
-                    var data = await response.Content.ReadAsStringAsync();
-                    var result =
-                        JsonConvert.DeserializeObject<NextApiResponseJsonWrapper<T>>(data, SerializationUtils.GetJsonConfig());
-                    if (!result.Success)
-                        throw NextApiClientUtils.NextApiException(result.Error);
-
-                    return result.Data;
-                }
-                case SerializationType.MessagePack:
+                throw NextApiClientUtils.NextApiException(result.Error);
+            }
+
+            return result.Data;
+        }
+
+        private async Task<T> InvokeHttpMessagePack<T>(HttpResponseMessage response)
+        {
+            var mediaType = response.Content.Headers.ContentType.MediaType;
+
+            if (typeof(T) == typeof(NextApiFileResponse) && !mediaType.Contains("application/json"))
+            {
+                try
                 {
-                    var resultByteArray = await response.Content.ReadAsByteArrayAsync();
-                    var result = (NextApiResponse) MessagePackSerializer.Typeless.Deserialize(resultByteArray);
-                    if (!result.Success)
-                        throw NextApiClientUtils.NextApiException(result.Error);
-                    return (T) result.Data;
+                    return await NextApiClientUtils.ProcessNextApiFileResponse(response) as dynamic;
                 }
-                default:
-                    throw new Exception($"Unsupported serialization type {HttpSerializationType}");
+                catch { }
```

Comment:
Should throw an exception with type `NextApiException`, like it been made in line 315.